app/components/Form/ProjectSummaryReportForm.tsx

```diff
@@ -1,13 +1,196 @@
+import { projectSummaryReportUiSchema } from "data/jsonSchemaForm/projectSummaryReportUISchema";
+import { JSONSchema7 } from "json-schema";
+import FormBase from "./FormBase";
+import { Button } from "@button-inc/bcgov-theme";
+import { ProjectSummaryReportForm_projectRevision$key } from "__generated__/ProjectSummaryReportForm_projectRevision.graphql";
+import { graphql, useFragment } from "react-relay";
+import { MutableRefObject, useRef, useMemo } from "react";
+import { useCreateProjectSummaryReport } from "mutations/ProjectSummaryReport/createProjectSummaryReport";
+import { useUpdateProjectSummaryReportFormChangeMutation } from "mutations/ProjectSummaryReport/updateProjectSummaryReportFormChange";
+import EmptyObjectFieldTemplate from "lib/theme/EmptyObjectFieldTemplate";
+import { useStageReportingRequirementFormChange } from "mutations/ProjectReportingRequirement/stageReportingRequirementFormChange";
+import stageMultipleReportingRequirementFormChanges from "./Functions/stageMultipleReportingRequirementFormChanges";
+import Status from "components/ReportingRequirement/Status";
+
 interface Props {
-  query: any;
-  projectRevision: any;
+  projectRevision: ProjectSummaryReportForm_projectRevision$key;
   onSubmit: () => void;
 }
+
 const ProjectSummaryReportForm: React.FC<Props> = (props) => {
+  const formRefs: MutableRefObject<{}> = useRef({});
+
+  const projectRevision = useFragment(
+    graphql`
+      fragment ProjectSummaryReportForm_projectRevision on ProjectRevision {
+        id
+        rowId
+        # eslint-disable-next-line relay/unused-fields
+        projectFormChange {
+          formDataRecordId
+        }
+        projectSummaryFormChanges: formChangesFor(
+          formDataTableName: "reporting_requirement"
+          reportType: "Project Summary Report"
+          first: 1000
+        ) @connection(key: "connection_projectSummaryFormChanges") {
+          edges {
+            node {
+              id
+              rowId
+              newFormData
+              changeStatus
+              formByJsonSchemaName {
+                jsonSchema
+              }
+            }
+          }
+        }
+        upcomingProjectSummaryReportFormChange: upcomingReportingRequirementFormChange(
+          reportType: "Project Summary Report"
+        ) {
+          id
+          # eslint-disable-next-line relay/must-colocate-fragment-spreads
+          ...ReportDueIndicator_formChange
+          asReportingRequirement {
+            reportDueDate
+          }
+        }
```

Comment:
The 'upcoming' requirement will always return the same one as the previous one, so you can add the fields / fragment spread to the `projectSummaryFormChanges` just above instead

Reply:
It looks like the this is the case after verifying with graphiql